plugins/stats_over_http/stats_over_http.cc

```diff
@@ -705,250 +718,209 @@ TSPluginInit(int argc, const char *argv[])
   }
 
   /* Create a continuation with a mutex as there is a shared global structure
      containing the headers to add */
-  main_cont = TSContCreate(stats_origin, NULL);
+  main_cont = TSContCreate(global_stats_origin_wrapper, NULL);
   TSContDataSet(main_cont, (void *)config_holder);
   TSHttpHookAdd(TS_HTTP_READ_REQUEST_HDR_HOOK, main_cont);
 
   /* Create continuation for management updates to re-read config file */
   config_cont = TSContCreate(config_handler, TSMutexCreate());
   TSContDataSet(config_cont, (void *)config_holder);
   TSMgmtUpdateRegister(config_cont, PLUGIN_NAME);
-  TSDebug(PLUGIN_NAME, "stats module registered with path %s", config_holder->config->stats_path);
+  TSDebug(PLUGIN_NAME, "stats module registered with path %s", config_holder->config->stats_path.c_str());
 
 done:
   return;
 }
 
-static bool
-is_ip_match(const char *ip, char *ipmask, char mask)
+///////////////////////////////////////////////////////////////////////////////
+// Initialize the plugin as a remap plugin.
+//
+TSReturnCode
+TSRemapInit(TSRemapInterface *api_info, char *errbuf, int errbuf_size)
 {
-  unsigned int j, i, k;
-  char cm;
-  // to be able to set mask to 128
-  unsigned int umask = 0xff & mask;
-
-  for (j = 0, i = 0; ((i + 1) * 8) <= umask; i++) {
-    if (ip[i] != ipmask[i]) {
-      return false;
-    }
-    j += 8;
-  }
-  cm = 0;
-  for (k = 0; j < umask; j++, k++) {
-    cm |= 1 << (7 - k);
+  if (api_info->size < sizeof(TSRemapInterface)) {
+    strncpy(errbuf, "[tsremap_init] - Incorrect size of TSRemapInterface structure", errbuf_size - 1);
+    return TS_ERROR;
   }
 
-  if ((ip[i] & cm) != (ipmask[i] & cm)) {
-    return false;
+  if (api_info->tsremap_version < TSREMAP_VERSION) {
+    snprintf(errbuf, errbuf_size, "[tsremap_init] - Incorrect API version %ld.%ld", api_info->tsremap_version >> 16,
+             (api_info->tsremap_version & 0xffff));
+    return TS_ERROR;
   }
-  return true;
+
+  TSDebug(PLUGIN_NAME, "remap plugin is successfully initialized");
+  return TS_SUCCESS;
 }
 
-static bool
-is_ip_allowed(const config_t *config, const struct sockaddr *addr)
+TSReturnCode
+TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf */, int /* errbuf_size */)
 {
-  char ip_port_text_buffer[INET6_ADDRSTRLEN];
-  int i;
-  char *ipmask;
-  if (!addr) {
-    return true;
+  config_holder_t *config_holder;
+  config_holder = new_config_holder(argc > 2 ? argv[2] : NULL);
+  *ih           = static_cast<void *>(config_holder);
+  /* Path was not set during load, so the param was not a config file, we also
+  have an argument so it must be the path, set it here.  Otherwise if no argument
+  then use the default _stats path */
+  if ((config_holder->config != NULL) && (config_holder->config->stats_path.empty()) && (argc > 2) &&
+      (config_holder->config_path == NULL)) {
+    config_holder->config->stats_path = TSstrdup(argv[2] + ('/' == argv[2][0] ? 1 : 0));
```

Comment:
Why not make `stats_path` a `std::string` and avoid calling not only `TSstrdup` but `TSfree`?